connect/basic-auth-extension/src/test/java/org/apache/kafka/connect/rest/basic/auth/extension/BasicAuthSecurityRestExtensionTest.java

```diff
@@ -63,4 +69,25 @@ public void testJaasConfigurationNotOverwritten() {
         assertNotEquals(overwrittenConfiguration, jaasFilter.getValue().configuration,
             "Overwritten JAAS configuration should not be used by basic auth REST extension");
     }
+
+    @Test
+    public void testBadJaasConfiguration() {
+        SecurityException jaasConfigurationException = new SecurityException(new IOException("Bad JAAS config is bad"));
+        Supplier<Configuration> configuration = BasicAuthSecurityRestExtension.initializeConfiguration(() -> {
+            throw jaasConfigurationException;
+        });
+
+        ConnectException thrownException = assertThrows(ConnectException.class, configuration::get);
+        assertEquals(jaasConfigurationException, thrownException.getCause());
+    }
+
+    @Test
+    public void testGoodJaasConfiguration() {
+        Configuration mockConfiguration = EasyMock.mock(Configuration.class);
```

Comment:
The identity function could pass this test, but wouldn't have the behavior we need in the BasicAuthSecurityRestExtension. I wonder if there's a way to confirm that the mockConfiguration has been evaluated prior to calling `get()` on the returned supplier.

Reply:
Not sure if you really meant evaluating `mockConfiguration`, or perhaps just validating that the wrapping `Supplier<Configuration>` was evaluated... I added a test case for the latter; LMKWYT.

Reply:
Another way is to use a mocked supplier and verify that the methods you expect to be called are indeed called (obviously only `get` here) and return or throw as you expect. This test does that, but with an additional level for indirection with this boolean, which basically implements what Easymock does already. Probably fine here since it results in more condensed code. But in the general case you'd pass a mock to verify that methods on that mock are called, which I also assume is referred to as evaluation here.